### PR TITLE
feat(inference): add full-IFU cube optimization objective layer

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add reusable IFU objective helpers in rubix.inference.objectives
- add optimize_ifu_cube wrapper that uses masked/weighted IFU cube losses
- export new APIs via rubix.inference package init
- add objective + integration unit tests
- document new objective module and optimize_ifu_cube usage

## Validation
- pre-commit run on changed files passed (black, isort, pydoclint)
- compileall passed for new/updated inference modules and tests

## Notes
- pytest execution in the local rubix conda environment did not return output in this sandbox session, so I validated with hooks + compile checks and included focused tests in this PR

### CODE REVIEW

File: rubix/inference/optimize.py [hunk 1]
```diff
@@ -225,3 +226,73 @@ def finished_step(finished_carry):
...
+
+
+def optimize_ifu_cube(
+    pipeline: Any,
+    params_init: ParamsTree,
+    static_data: RubixData,
+    target: jnp.ndarray,
+    mask: Optional[jnp.ndarray] = None,
+    weights: Optional[jnp.ndarray] = None,
+    normalize_loss: bool = True,
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/optimize.py:238]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/208#discussion_r3113107015)

File: rubix/inference/optimize.py [hunk 2]
```diff
@@ -225,3 +226,73 @@ def finished_step(finished_carry):
...
+        OptimizationResult: Standard optimization traces and best/final params.
+    """
+    if target.ndim != 3:
+        raise ValueError("target must be a 3D IFU datacube")
+
+    cube_loss_fn = build_ifu_cube_loss(
+        mask=mask,
+        weights=weights,
+        normalize=normalize_loss,
+    )
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply rubix/inference/optimize.py:297]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/208#discussion_r3113107038)
